NLog.Web.AspNetCore/LayoutRenderers/AspNetRequestPostedBody.cs

```diff
@@ -47,75 +50,108 @@ protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
             {
                 return;
             }
-
-            long? contentLength = httpRequest.ContentLength;
-            if (contentLength <= 0)
+          
+            if (TryGetBody(httpRequest, out var body))
             {
                 return; // No Body to read
             }
 
-            if (MaxContentLength > 0 && contentLength > MaxContentLength)
+            // reset if possible
+            if (TryResetStream(httpRequest, out var oldPosition, ref body))
             {
-                InternalLogger.Debug("AspNetRequestPostedBody: body stream is too big. ContentLength={0}", contentLength);
-                return; // Body too large
+                return;
             }
+            
+            var content = BodyToString(body);
 
-#if !ASP_NET_CORE
-            var body = httpRequest.InputStream;
-#else
-            var body = httpRequest.Body;
-#endif
+            //restore
+            body.Position = oldPosition;
 
-            if (body == null)
-            {
-                InternalLogger.Debug("AspNetRequestPostedBody: body stream was null");
-                return;
-            }
+            builder.Append(content);
+        }
 
-            if (!body.CanRead)
-            {
-                InternalLogger.Debug("AspNetRequestPostedBody: body stream has been closed");
-                return;
-            }
+        private static string BodyToString(Stream body)
+        {
+            //note: dispose of StreamReader isn't doing things besides closing the stream (which can be turn off, and then it's a NOOP)
```

Comment:
Think the correct comment is that StreamReader is NOT disposed because it would close the stream, which is not wanted.

Reply:
that's what is says? But will reword it

Reply:
done

Reply:
And there is no finalizer on StreamReader, so it will not be magically closed either.